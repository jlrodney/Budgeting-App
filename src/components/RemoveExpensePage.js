import React from 'react';
import { connect } from 'react-redux';
import { startRemoveExpense } from '../actions/expenses';

export class RemoveExpensePage extends React.Component {
  onRemove = () => {
    this.props.startRemoveExpense({id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        Are you sure you want to delete "{this.props.expense.description}" expense
        <button onClick={this.onRemove}>Yes! Delete.</button>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startRemoveExpense: (id, expense) => dispatch(startRemoveExpense(id, expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(RemoveExpensePage);
