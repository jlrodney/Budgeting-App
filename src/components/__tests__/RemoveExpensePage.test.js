import React from 'react';
import { shallow } from 'enzyme';
import { RemoveExpensePage } from '../RemoveExpensePage';

let removeExpense, history, wrapper;
const expenses = [{
  id: '1',
  description: 'Elictricity',
  note: '',
  amount: 180,
  createdAt: 0
}]
beforeEach(() => {
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <RemoveExpensePage
      removeExpense={removeExpense}
      history={history}
      expense={expenses[0]}
    />
  );
});
describe('RemoveExpensePage', () => {
  test('should render RemoveExpensePage', () => {
    expect(wrapper).toMatchSnapshot();
  });


  test('should handle removeExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({
      id: expenses[0].id
    });
  });
});
