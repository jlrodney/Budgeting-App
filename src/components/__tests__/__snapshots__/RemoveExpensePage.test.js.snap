// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveExpensePage should render RemoveExpensePage 1`] = `
<div>
  Are you sure you want to delete "
  Elictricity
  " expense
  <button
    onClick={[Function]}
  >
    Yes! Delete.
  </button>
</div>
`;
